genesyscloud/telephony_provider_edges_trunkbasesettings/resource_genesyscloud_telephony_providers_edges_trunkbasesettings.go

```
@@ -212,33 +160,45 @@ func updateTrunkBaseSettings(ctx context.Context, d *schema.ResourceData, meta i
 	trunkBase.Version = trunkBaseSettings.Version
 
 	log.Printf("Updating trunk base settings %s", name)
-	trunkBaseSettings, resp, err := edgesAPI.PutTelephonyProvidersEdgesTrunkbasesetting(d.Id(), trunkBase)
+	trunkBaseSettings, resp, err := proxy.UpdateTrunkBaseSetting(ctx, d.Id(), trunkBase)
 	if err != nil {
 		return util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to update trunk base settings %s error: %s", d.Id(), err), resp)
 	}
 
-	log.Printf("Updated trunk base settings %s", *trunkBaseSettings.Id)
+	if trunkBaseSettings != nil && trunkBaseSettings.Id != nil {
+		log.Printf("Updated trunk base settings %s", *trunkBaseSettings.Id)
+	} else {
+		log.Printf("Error: trunkBaseSettings or its Id is nil")
```

Should we return this error?